app/javascript/src/warn-dsit-organisation-change.js

```diff
@@ -0,0 +1,40 @@
+/*
+  Show a standard JavaScript `confirm` modal with a warning message if a new
+  user is being created whose email address is not in the DSIT whitelist and
+  their organisation is being set to DSIT, or when an existing user is being
+  edited whose email address is not in the DIST whitelist and their
+  organisation is being set to DSIT from a different organisation.
+*/
+document.addEventListener("DOMContentLoaded", function() {
+  const VALID_DSIT_DOMAINS = ["dsit.gov.uk", "odamanagement.org"];
+  const DSIT_ORGANISATION_INDEX = 0;
+  const WARNING = "Are you sure you wish to add this user's primary organisation as DSIT?";
```

Comment:
We should probably say why this is a warning to give a clue as to what to do next. For example
```suggestion
  const WARNING = "This user does not appear to be from dsit.gov.uk or odamanagement.org. Are you sure you wish to add this user's primary organisation as DSIT?";
```